operator/controllers/kyma_controller.go

```diff
@@ -201,6 +205,21 @@ func (r *KymaReconciler) HandleProcessingState(ctx context.Context, kyma *v1alph
 			fmt.Errorf("error while syncing conditions during processing: %w", err))
 	}
 
+	notExistsModuleInfos := kyma.GetNotExistsModuleInfos()
+	if len(notExistsModuleInfos) > 0 {
+		requireSync, err := r.DeleteNotExistsModulesAndUpdateConditions(ctx, &kyma.Status, notExistsModuleInfos)
+		if err != nil {
+			return r.UpdateStatusFromErr(ctx, kyma, v1alpha1.KymaStateError,
+				fmt.Errorf("error while syncing conditions during deleting non exists modules: %w", err))
+		}
+		if requireSync {
+			if err := status.Helper(r).UpdateStatusForExistingModules(
```

Comment:
after we update a status we should requeue again to not run into inconsistency errors

Reply:
refactored